src/java.base/share/classes/java/lang/System.java

```diff
@@ -2226,7 +2226,23 @@ private static void initPhase1() {
         SharedSecrets.getJavaLangRefAccess().startThreads();
 
         // system properties, java.lang and other core classes are now initialized
-        VM.initLevel(1);
+        VM.initLevel(VM.JAVA_LANG_SYSTEM_INITED);
+
+        try {
+            initJSR292Classes();
+        } catch (Throwable ex) {
+            throw new InternalError(ex);
+        }
+        VM.initLevel(VM.JAVA_LANG_INVOKE_INITED);
+    }
+
+    // do phase1 initializations specific to JSR 292 APIs
+    private static void initJSR292Classes() throws Throwable {
+        var mh = java.lang.invoke.MethodHandles.identity(int.class);
+        int z = 0;
+        z |= (int) mh.invokeExact(z);
+        z |= (Integer) mh.invoke((Object)0);
```

Comment:
As you saw in the new version, I added some more API exercise.

It is pretty arbitrary, but the goal is to ensure that both kinds of invocation work.  The untyped invocation will cause an asType call to be performed.

The specific types and arities are unimportant.  This approach won't populate many caches, but it will touch and init lots of key jli classes.

A more thorough approach might set up a goal-set of classes to load, and then tune the ad hoc code so that it is a short as possible, while still touching the goal-set.  But it would be fragile, assuming that indirectly touched classes would be touched even after refactoring happens elsewhere.

I think my next attempt will make sure to mention all the jli classes in my goal-set, and make sure each one is touched in a sure-fire way by a bit of code.  Then I can start to delete unnecessary bits.